starlette/middleware/base.py

```diff
@@ -42,18 +48,42 @@ async def coro() -> None:
         task = loop.create_task(coro())
         message = await queue.get()
         if message is None:
+            queue.task_done()
             task.result()
             raise RuntimeError("No response returned.")
         assert message["type"] == "http.response.start"
 
         async def body_stream() -> typing.AsyncGenerator[bytes, None]:
-            while True:
+            def streaming_predicate(
+                msg: typing.Optional[dict], more_body: bool = True
+            ) -> bool:
+                return (
+                    msg is not None
+                    and msg["type"] == "http.response.body"
+                    and "more_body" in msg
+                    and msg["more_body"] is more_body
+                )
+
+            # In non-streaming responses, there will be one message to emit
+            message = await queue.get()
+            if message is not None:
+                queue.task_done()
+                assert message["type"] == "http.response.body"
+                yield message.get("body", b"")
+
+            while streaming_predicate(message, more_body=True):
```

Comment:
If you indented this while loop by one extra step, could you just do:
```python
while message.get("more_body"):
   ...
```

Reply:
Oh, I see: under the first check if the message is `None`. Yes, that should work.